contracts/utils/RegistryBootstrap.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+// ──────────────────────────────────────────────────────────────────────────────
+//     _   __    _  __
+//    / | / /__ | |/ /_  _______
+//   /  |/ / _ \|   / / / / ___/
+//  / /|  /  __/   / /_/ (__  )
+// /_/ |_/\___/_/|_\__,_/____/
+//
+// ──────────────────────────────────────────────────────────────────────────────
+// Nexus: A suite of contracts for Modular Smart Accounts compliant with ERC-7579 and ERC-4337, developed by Biconomy.
+// Learn more at https://biconomy.io. For security issues, contact: security@biconomy.io
+
+import { ModuleManager } from "../base/ModuleManager.sol";
+import { IModule } from "../interfaces/modules/IModule.sol";
+import { IERC7484 } from "../interfaces/IERC7484.sol";
+
+/// @title Bootstrap Configuration for Nexus
+/// @notice Provides configuration and initialization for Nexus smart accounts.
+/// @author @livingrockrises | Biconomy | chirag@biconomy.io
+/// @author @aboudjem | Biconomy | adam.boudjemaa@biconomy.io
+/// @author @filmakarov | Biconomy | filipp.makarov@biconomy.io
+/// @author @zeroknots | Rhinestone.wtf | zeroknots.eth
+/// Special thanks to the Solady team for foundational contributions: https://github.com/Vectorized/solady
+struct BootstrapConfig {
+    address module;
+    bytes data;
+}
+
+/// @title Bootstrap
+/// @notice Manages the installation of modules into Nexus smart accounts using delegatecalls.
+contract Bootstrap is ModuleManager {
+    /// @notice Initializes the Nexus account with a single validator.
+    /// @dev Intended to be called by the Nexus with a delegatecall.
+    /// @param validator The address of the validator module.
+    /// @param data The initialization data for the validator module.
+    function initNexusWithSingleValidator(
+        IModule validator,
+        bytes calldata data,
+        IERC7484 registry,
+        address[] calldata attesters,
+        uint8 threshold
+    )
+        external
+    {
+        _installValidator(address(validator), data);
+        _configureRegistry(registry, attesters, threshold);
+    }
+
+    /// @notice Initializes the Nexus account with multiple modules.
+    /// @dev Intended to be called by the Nexus with a delegatecall.
+    /// @param validators The configuration array for validator modules.
+    /// @param executors The configuration array for executor modules.
+    /// @param hook The configuration for the hook module.
+    /// @param fallbacks The configuration array for fallback handler modules.
+    function initNexus(
```

Comment:
is it okay to have two kinds of methods here withRegistry and withoutRegistry
and similarly K1ValidatorFactory and K1ValidatorFactoryWithRegistry 

if I use K1ValidatorFactoryWithRegistry with 0 addresses then it would unnecessarily increase gas. we can do comparison benchmark though @Aboudjem